trieste/acquisition/rule.py

```diff
@@ -2073,6 +2073,183 @@ def update(
         pass
 
 
+class SingleObjectiveTrustRegionDiscrete(UpdatableTrustRegionDiscrete):
+    """
+    An updatable discrete trust region that maintains a set of neighboring points around a
+    single location point. The region is updated based on the best point found in the region.
+    """
+
+    def __init__(
+        self,
+        global_search_space: DiscreteSearchSpace,
+        beta: float = 0.7,
+        kappa: float = 1e-4,
+        min_eps: float = 1e-2,
+        region_index: Optional[int] = None,
+        input_active_dims: Optional[Union[slice, Sequence[int]]] = None,
+    ):
+        """
+        Select a random initial location from the global search space and select the initial
+        neighbors within the trust region.
+
+        :param global_search_space: The global search space this search space lives in.
+        :param beta: The inverse of the trust region contraction factor.
+        :param kappa: Scales the threshold for the minimal improvement required for a step to be
+            considered a success.
+        :param min_eps: The minimal size of the search space. If the size of the search space is
+            smaller than this, the search space is reinitialized.
+        :param region_index: The index of the region in a multi-region search space. This is used to
+            identify the local models and datasets to use for acquisition. If `None`, the
+            global models and datasets are used.
+        :param input_active_dims: The active dimensions of the input space, either a slice or list
+            of indices into the columns of the space. If `None`, all dimensions are active.
+        """
+        super().__init__(global_search_space, region_index, input_active_dims)
+        self._beta = beta
+        self._kappa = kappa
+        self._min_eps = min_eps
+        self._initialized = False
+        self._step_is_success = False
+
+        # Random initial location index from the global search space.
+        self._location_ix = tf.random.categorical(
+            tf.ones(
+                (
+                    1,
+                    global_search_space.points.shape[0],
+                )
+            ),
+            1,
+        )[0]
+
+        # Pairwise distances for each axis in the global search space.
+        points = global_search_space.points
+        self._global_distances = tf.abs(tf.expand_dims(points, -2) - tf.expand_dims(points, -3))
+
+        self._init_eps()
+        self._update_neighbors()
+        self._y_min = np.inf
+
+    @property
+    def location(self) -> TensorType:
+        """Center point of the region."""
+        return tf.reshape(tf.gather(self.global_search_space.points, self._location_ix), (-1,))
+
+    def _init_eps(self) -> None:
+        global_lower = self.global_search_space.lower
+        global_upper = self.global_search_space.upper
+        self.eps = 0.5 * (global_upper - global_lower) / (5.0 ** (1.0 / global_lower.shape[-1]))
```

Comment:
we should parametrize the multiplier and update the eps rule, this one is outdated,  we overlooked this previously....

Reply:
can you raise a ticket to come back to this in a separate PR?

Reply:
I can add a parameter as part of this PR, e.g. `zeta` can default to `0.5`, but what about the divisor?

Reply:
I guess we also want to change `SingleObjectiveTrustRegionBox`, so perhaps it makes sense to do it in a separate PR like you suggest.

Reply:
PR #832 addresses this.